python/python/bystro/prs/prs_listener.py

```diff
@@ -0,0 +1,150 @@
+"""Provide a worker for PRS calculation."""
+
+import argparse
+import logging
+from pathlib import Path
+
+from bystro.beanstalkd.messages import BaseMessage, CompletedJobMessage, SubmittedJobMessage
+from bystro.beanstalkd.worker import ProgressPublisher, QueueConf, get_progress_reporter, listen
+from bystro.utils.timer import Timer
+
+# TODO: Define PRS calculation and preprocess covariate, association files
+from bystro.prs.prs_types import PRSResponse, PRSSubmission
+from bystro.prs.calculate_prs import (
+    _load_dosage_matrix,
+    _load_association_scores,
+    _load_covariates,
+    calculate_prs_scores,
+)
+
+# TODO: Define how we load vcf-like genotype matrix from annotation,
+# then generalize this process for other analyses
+from bystro.prs.utils import load_vcf_from_annotation
+
+# TODO: Implement function to preprocess genotype data from annotation
+# into format suitable for PRS calculation
+from bystro.prs.utils import preprocess_vcf_for_prs
+
+from ruamel.yaml import YAML
+
+logging.basicConfig(
+    filename="prs_worker.log",
+    level=logging.DEBUG,
+    format="%(asctime)s.%(msecs)03d %(levelname)s %(module)s - %(funcName)s: %(message)s",
+    datefmt="%Y-%m-%d %H:%M:%S",
+)
+logger = logging.getLogger()
+
+PRS_TUBE = "prs_analysis"
+
+
+class PRSJobData(BaseMessage, frozen=True):
+    """Wrap a PRSSubmission in a BaseMessage for beanstalkd."""
+
+    prs_submission: PRSSubmission
+
+
+class PRSJobCompleteMessage(CompletedJobMessage, frozen=True, kw_only=True):
+    """Wrap a PRSResponse in a CompletedJobMessage for beanstalkd."""
+
+    results: PRSResponse
+
+
+def _load_queue_conf(queue_conf_path: str) -> QueueConf:
+    with Path(queue_conf_path).open(encoding="utf-8") as queue_config_file:
+        raw_queue_conf = YAML(typ="safe").load(queue_config_file)
+    beanstalk_conf = raw_queue_conf["beanstalkd"]
+    return QueueConf(addresses=beanstalk_conf["addresses"], tubes=beanstalk_conf["tubes"])
+
+
+def handler_fn(publisher: ProgressPublisher, prs_job_data: PRSJobData) -> PRSResponse:
```

Comment:
Recommend that you add a stub/dummy PRSResponse here, and add a TODO to finish it off. And then in this PR (I can help) add a test demonstrating that the server works.